### ISSUE TITLE
Change Vendor- and Product-ID etg.

### ISSUE THREAD

**Issue by AintLarry** [jakubdybczak/esp32-arduino-matter-builder#1]
Hey,

I know, that the "factory-data" ist precompiled in your library. 

Do you know, where I can change the factory_partition.bin? Do I have to use the ESP-Matter/mfg-tool oder factory_data provider?

Or is it possible to changes these values somewhere else before building the lib?

Thank you for your help.

Henrik

**Comment by Massimiliano-solutiontech** [comment]
+1

**Comment by Yacubane** [comment]
>Do I have to use the ESP-Matter/mfg-tool oder factory_data provider?

yes, it looks like, according to https://docs.espressif.com/projects/esp-matter/en/latest/esp32/developing.html#using-esp-secure-cert-partition

>Or is it possible to changes these values somewhere else before building the lib?

I'm not aware of such option, I would suggest trying using factory data provider

**Comment by Yacubane** [comment]
In newest version of this lib (beta 5) there should be a possibility to change this PIN. Please look here:
https://github.com/espressif/esp-matter/blob/4707b88ceb5e6328f7b571b652487507429c1891/docs/en/developing.rst (2.5 Factory Data Providers section)

I have enabled custom providers for every type of provider in that build.

**Comment by norberthidas** [comment]
Hi
I am setting up a device with temperature measurement runnig from battery. It is running on esp32. I can send a temperature and can see it in google home app, but I want to send the battery percentage to the google home app too, but with no luck till now. How can I do that? Can you tell me some info about reporting battery state from matter device?

Thank you in advance.
